src/cython_code/State.pyx

```diff
@@ -395,7 +404,12 @@ cdef class p_State:
                                  'State.transition: %s' % which_transition
                              print(print_str)
                    else:
-                       continue
+                        step_idx += 1
+                        continue
+                   if progress:
+                     print(
+                       '\rCompleted: %d iterations in %f seconds.' %
+                       (step_idx, elapsed_secs))
```

Comment:
I also need `progress` to receive the total number of iterations and elapsed seconds from `transition`.